toolkit/tools/internal/rpm/rpm.go

```diff
@@ -89,8 +91,48 @@ var (
 	//
 	//	D: ========== +++ systemd-devel-239-42.azl3 x86_64-linux 0x0
 	installedRPMRegex = regexp.MustCompile(`^D: =+ \+{3} (\S+) (\S+)-linux.*$`)
+
+	// For most use-cases, the distro name abbreviation and major version are set by the exe package. However, if the
+	// module is used outside of the main Mariner build system, the caller can override these values with SetDistroMacros().
+	distNameAbreviation, distMajorVersion = loadLdDistroFlags()
 )
 
+// checkDistroMacros validates the distro macro values.
+func checkDistroMacros(nameAbreviation string, majorVersion int) (string, int, error) {
+	if majorVersion < 0 || nameAbreviation == "" {
+		err := fmt.Errorf("failed to set distro defines (%s, %d), invalid name or version", nameAbreviation, majorVersion)
+		return "", 0, err
+	}
+	return nameAbreviation, majorVersion, nil
```

Comment:
We're not setting or updating these values here, so no need to return them. I think we might as well completely drop this function and just move the `if majorVersion < 0 || nameAbreviation == ""` directly to the caller, since there's only one. Alternatively, the function could return a boolean and leave it to the caller to construct a proper error message.

Reply:
Both `loadLdDistroFlags()` and `SetDistroMacros()` call it, but I agree it doesn't need to return the values (hold over from when it was calling atoi).